### PR TITLE
iox-#1196 Fix warnings in access control

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Part of #1196

### CODE REVIEW

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/access_control.hpp [hunk 1]
```diff
@@ -88,11 +93,14 @@ class AccessController
...
     ///   Category::OTHER the f_id is not required for everything else a valid group or user id is mandatory.
-    bool
-    addPermissionEntry(const Category f_category, const Permission f_permission, const uint32_t f_id = -1u) noexcept;
+    bool addPermissionEntry(const Category f_category,
+                            const Permission f_permission,
+                            const uint32_t f_id = INVALID_ID) noexcept;
 
     /// @brief just like addPermissionEntry(Category, Permission, int) but using a name instead of an id.
-    bool addPermissionEntry(const Category f_category, const Permission f_permission, const string_t& f_name) noexcept;
+    bool addPermissionEntry(const Category f_category,
```
Comment (mossmaurice) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/access_control.hpp:101]: Could you remove the silly `f_` everywhere is this file? :smile_cat: I would appreciate that a lot :upside_down_face: 
```suggestion
    bool addPermissionEntry(const Category category,
```

File: iceoryx_hoofs/source/posix_wrapper/access_control.cpp [hunk 2]
```diff
@@ -94,15 +94,19 @@ AccessController::createACL(const int32_t f_numEntries) noexcept
     cxx::function<void(acl_t)> freeACL = [&](acl_t acl) {
         auto aclFreeCall = posixCall(acl_free)(acl).successReturnValue(0).evaluate();
         // We ensure here instead of returning as this lambda will be called by unique_ptr
+        /// NOLINTJUSTIFICATION will be replaced with refactored error handling
```
Comment (mossmaurice) [inline iceoryx_hoofs/source/posix_wrapper/access_control.cpp:97]: Nitpick: You could add the issue number and a todo here.

File: iceoryx_hoofs/test/moduletests/test_access_control.cpp [hunk 3]
```diff
@@ -15,14 +15,12 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(unix) && !defined(__unix) && !defined(__unix__)
+#if defined(__linux__)
```
Comment (mossmaurice) [inline iceoryx_hoofs/test/moduletests/test_access_control.cpp:18]: :rocket:

File: iceoryx_hoofs/test/moduletests/test_access_control.cpp [hunk 4]
```diff
@@ -251,9 +248,9 @@ TEST_F(AccessController_test, addNameInWrongPlace)
 TEST_F(AccessController_test, addManyPermissions)
 {
     ::testing::Test::RecordProperty("TEST_ID", "998c828b-8b9e-4677-9c36-4a1251c11241");
-    AccessController::string_t groupName = "root";
+    AccessController::permissionString_t groupName = "root";
 
-    bool entryAdded;
+    bool entryAdded = false;
```
Comment (mossmaurice) [inline iceoryx_hoofs/test/moduletests/test_access_control.cpp:253]: AUTOSAR prefers 
```suggestion
    bool entryAdded{false};
```
If we change it on this PR, we won't have to care about it anymore. Same for the other variable above.

**Review by mossmaurice — COMMENTED** [review]
The changes LGTM. A few small things.

File: iceoryx_hoofs/test/moduletests/test_access_control.cpp [hunk 5]
```diff
@@ -15,55 +15,77 @@
...
         if (Test::HasFailure())
         {
             std::cout << output << std::endl;
         }
     }
 
-    ~AccessController_test()
+    iox::posix::AccessController m_accessController;
+    FILE* m_fileStream = nullptr;
+    int m_fileDescriptor = 0;
```
Comment (mossmaurice) [inline iceoryx_hoofs/test/moduletests/test_access_control.cpp:55]: Nitpick as this is only a test: Always try to prefer brace initializers.

**Review by mossmaurice — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Pull request merged**